client/app/scripts/superdesk/notification/notification.js

```diff
@@ -29,6 +44,102 @@
+        };
+
+        $rootScope.$on('reload', function(event, msg) {
+            _this.result = _this.reloadIdentifier(msg);
+            _this.reload(_this.result);
+        });
+        this.reload = function(result) {
+            if (result.reload) {
+                if ($window.location.hash != null && $window.location.hash.match('/authoring/') != null) {
+                    _this.broadcast(result.message);
+                } else {
+                    $window.location.reload(true);
+                }
+            }
+        };
+
+        this.broadcast = function(msg) {
+            $rootScope.$broadcast('savework', msg);
+        };
+
+        this.reloadIdentifier = function(msg) {
+            var result = {
+                reload: false,
+                message: null
+            };
+            if (_.has(userEvents, msg.event)) {
+                if (msg.extra.user_id != null) {
+                    if (msg.extra.user_id.indexOf(session.identity._id)  !== -1) {
+                        result.message = userEvents[msg.event];
+                        result.reload = true;
+                    }
+                }
+            }
+
+            if (_.has(roleEvents, msg.event)) {
+                if (msg.extra.role_id.indexOf(session.identity.role)  !== -1) {
+                    result.message = roleEvents[msg.event];
+                    result.reload = true;
+                }
+            }
+            if (_.has(deskEvents, msg.event)) {
+                if (msg.extra.desk_id != null && msg.extra.user_ids != null) {
+                    if (_.find(_this.userDesks, {_id: msg.extra.desk_id}) != null &&
+                        msg.extra.user_ids.indexOf(session.identity._id)  !== -1) {
+                        result.message = deskEvents[msg.event];
+                        result.reload = true;
+                    }
+                }
+            }
+
+            if (_.has(stageEvents, msg.event)) {
+                if (msg.extra.desk_id != null) {
+                    if (msg.event === 'stage_visibility_updated') {
+                        if (_.find(_this.userDesks, {_id: msg.extra.desk_id}) == null &&
+                        ($window.location.hash.match('/search') != null || $window.location.hash.match('/authoring/') != null)) {
+                            result.message = stageEvents[msg.event];
+                            result.reload = true;
+                        }
+                    } else if (msg.event === 'stage') {
+                        if (_.find(_this.userDesks, {_id: msg.extra.desk_id}) != null) {
+                            result.message = stageEvents[msg.event];
+                            result.reload = true;
+                        }
+                    }
+                }
+            }
+            return result;
+        };
+    }
+
+    return angular.module('superdesk.notification', ['superdesk.desks'])
+        .service('reloadService', ReloadService)
```

Comment:
Since there are other variables and event using in this work with the name 'reload', that's why using 'reloadService' as service name here.